### PR TITLE
Fix Shaman Totems and Elementals

### LINKED ISSUE #87: Game - Shaman Totems in Duels
**Description:**

Shaman Totems are currently friendly during /duels with players of the same faction, thus not attacking the target, etc.

**Current behaviour:**

Shaman Totems are currently friendly during /duels with players of the same faction, thus not attacking the target, etc.

**Expected behaviour:**

Summoning a totem during any duel should go hostile towards that player when the duel begins.

**Steps to reproduce the problem:**

1. Create a Shaman.
2. Create another character of the same faction.
3. Cast any or all Totem(s).

**TC rev. hash/commit:** 
TrinityCore rev. 04b99ab8ac2f0730a7e652945daac1c5e5211175

**TDB version:**  
434.20021

**Operating system:** 
Win 10 (x64)

### LINKED ISSUE #213: [SPELL/SHAMAN] Totems friendly / hostile not reflecting casters state
**Current behaviour:**

image
(Totem attacking in Dalaran which is a sanctuary)

**Expected behaviour:**

Shaman totems should be attackable when the shaman is. So when the shaman is in a sanctuary, where you can not attack the opposing faction, his totems should also not be attackable/attack.

Additionally, when dueling a shaman of same faction, you can not attack his totems.
image

**Steps to reproduce the problem:**

Check the screenshots.

**TC rev. hash/commit:** 

4a158fdf22089a3fffacc9df19b814c26452e658

**TDB version:** 

TDB 434.20091

**Operating system:** 

Win10 64

### PR DESCRIPTION
**Pull request by AGandrup**
**Changes proposed**:

- Totems and Elementals will now properly engage in both PvE and PvP combat.
- Totems and Elementals are now properly affected by sanctuary and FFA areas as well as dueling state.
- Elementals will now aggro when its owner is attacked.

**Issues addressed**: 

- Fixes #87 
- Fixes #213

**Tests performed**: 

- Builds
- Tested in duels both same faction and cross faction, contested area, sanctuary area (Dalaran Sewers), FFA PvP area (Gurubashi Arena). Works as intended.

**Known issues and TODO list**:

- None from what I can tell.

### CODE REVIEW

File: src/server/game/Entities/Totem/Totem.cpp [hunk 1]
```diff
@@ -66,6 +66,11 @@ void Totem::InitStats(uint32 duration)
 
         // set display id depending on caster's race
         SetDisplayId(GetOwner()->GetModelForTotem(PlayerTotemType(m_Properties->ID)));
+
+        SetOwnerGUID(GetOwner()->GetGUID());
```
Comment (Ovahlord) [inline src/server/game/Entities/Totem/Totem.cpp:70]: hack. Totems are not 'owned'. this applies only for charmed units

Comment (AGandrup) [reply src/server/game/Entities/Totem/Totem.cpp:70]: Ye, I kind of figured that this wouldnt be the right way considering that other places in the code it is said not to call SetMinion method on totems. 

Nonetheless, I can say that right now, without setting owner guid, totems wont attack same faction in duels.

Comment (Ovahlord) [reply src/server/game/Entities/Totem/Totem.cpp:70]: yes, which is why I keep it unfixed. I wont tuctape stuff for the sake of ductaping it.

Comment (AGandrup) [reply src/server/game/Entities/Totem/Totem.cpp:70]: Right, thats understandable. I have no idea, so I just try to fix stuff, make a PR and see what happens 😝

**Pull request closed without merging**